src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/amendhearingoutcome/AmendHearingOutcomeAboutToSubmitHandler.java

```diff
@@ -31,12 +37,62 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
             throw new IllegalStateException("Cannot handle callback");
         }
 
-        SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        List<String> hearingsSelected = new ArrayList<>();
 
         if (sscsCaseData.getHearingOutcomes().isEmpty()) {
             sscsCaseData.setHearingOutcomes(null);
+            return new PreSubmitCallbackResponse<>(sscsCaseData);
+        } else {
+            for (HearingOutcome selectedHearingOutcome :  sscsCaseData.getHearingOutcomes()) {
+                String selectedHearingId =
+                        selectedHearingOutcome.getValue().getCompletedHearings().getValue().getCode();
+                if (hearingsSelected == null) {
+                    hearingsSelected.add(selectedHearingId);
+                } else if (hearingsSelected.contains(selectedHearingId)) {
+                    log.info("Hearing outcome {} selected more than once for case {}",
+                            selectedHearingOutcome.getValue().getCompletedHearings().getValue(),
+                            caseDetails.getId());
+                    preSubmitCallbackResponse.addError("This hearing already has an outcome recorded: "
+                            + selectedHearingOutcome.getValue().getCompletedHearings().getValue().getLabel());
+                    return preSubmitCallbackResponse;
+                } else {
+                    hearingsSelected.add(selectedHearingId);
+                }
+            }
+        }
+
+        int hearingOutcomesSize = sscsCaseData.getHearingOutcomes().size();
+
+        for (HearingOutcome hearingOutcome : sscsCaseData.getHearingOutcomes()) {
```

Comment:
this looks like the same for loop as the one above. could we move the logic to the above for loop?